### PR TITLE
Add table aws_eks_identity_provider_config. Closes #549

### LINKED ISSUE #549: Add table aws_eks_identity_provider_config
**References**
https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/eks_identity_provider_config

### PR DESCRIPTION
**Pull request by ParthaI**
# Integration test logs
<details>
  <summary>Logs</summary>
  
```
SETUP: tests/aws_eks_identity_provider_config []

PRETEST: tests/aws_eks_identity_provider_config

TEST: tests/aws_eks_identity_provider_config
Running terraform
aws_iam_role.named_test_resource: Creating...
aws_vpc.named_test_resource: Creating...
aws_iam_role.named_test_resource: Creation complete after 5s [id=turbottest32066]
aws_iam_role_policy_attachment.named_test_resource: Creating...
aws_iam_role_policy_attachment.named_test_resource2: Creating...
aws_iam_role_policy_attachment.named_test_resource: Creation complete after 2s [id=turbottest32066-20210728103244897700000001]
aws_iam_role_policy_attachment.named_test_resource2: Creation complete after 2s [id=turbottest32066-20210728103244933300000002]
aws_vpc.named_test_resource: Still creating... [10s elapsed]
aws_vpc.named_test_resource: Creation complete after 13s [id=vpc-06d7d1072653b6684]
aws_subnet.named_test_resource2: Creating...
aws_subnet.named_test_resource1: Creating...
aws_subnet.named_test_resource1: Creation complete after 3s [id=subnet-0227413530d53ca0a]
aws_subnet.named_test_resource2: Creation complete after 3s [id=subnet-0bbeb91381af36999]
aws_eks_cluster.named_test_resource: Creating...
aws_eks_cluster.named_test_resource: Still creating... [11s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [21s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [31s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [41s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [51s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [1m1s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [1m11s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [1m21s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [1m31s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [1m41s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [1m51s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [2m1s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [2m11s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [2m21s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [2m31s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [2m41s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [2m51s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [3m1s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [3m11s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [3m21s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [3m31s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [3m41s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [3m51s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [4m1s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [4m11s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [4m21s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [4m31s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [4m41s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [4m51s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [5m1s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [5m11s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [5m21s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [5m31s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [5m41s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [5m51s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [6m1s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [6m11s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [6m21s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [6m31s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [6m41s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [6m51s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [7m1s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [7m11s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [7m21s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [7m31s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [7m41s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [7m51s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [8m1s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [8m11s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [8m21s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [8m31s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [8m41s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [8m51s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [9m1s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [9m11s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [9m21s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [9m31s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [9m41s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [9m51s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [10m1s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [10m11s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [10m21s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [10m31s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [10m41s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [10m51s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [11m1s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [11m11s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [11m21s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [11m31s elapsed]
aws_eks_cluster.named_test_resource: Creation complete after 11m35s [id=turbottest32066]
aws_eks_identity_provider_config.named_test_resource: Creating...
aws_eks_identity_provider_config.named_test_resource: Still creating... [10s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [20s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [30s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [40s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [50s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [1m0s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [1m10s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [1m20s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [1m30s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [1m40s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [1m50s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [2m0s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [2m10s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [2m20s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [2m30s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [2m40s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [2m50s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [3m0s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [3m10s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [3m20s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [3m30s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [3m40s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [3m50s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [4m0s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [4m10s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [4m20s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [4m30s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [4m40s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [4m50s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [5m0s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [5m10s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [5m20s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [5m30s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [5m40s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [5m50s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [6m0s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [6m10s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [6m20s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [6m30s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [6m40s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [6m50s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [7m0s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [7m10s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [7m20s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [7m30s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [7m40s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [7m50s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [8m0s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [8m10s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [8m20s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [8m30s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [8m40s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [8m50s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [9m0s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [9m10s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [9m20s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [9m30s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [9m40s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [9m50s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [10m0s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [10m10s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [10m20s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [10m30s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [10m40s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [10m50s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [11m0s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [11m10s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [11m20s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [11m30s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [11m40s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [11m50s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [12m0s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [12m10s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [12m20s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [12m30s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [12m40s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [12m50s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [13m0s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [13m10s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [13m20s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [13m30s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [13m40s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [13m50s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [14m0s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [14m10s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [14m20s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [14m30s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [14m40s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [14m50s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [15m0s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [15m10s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [15m20s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [15m30s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [15m40s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [15m50s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [16m0s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [16m10s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [16m20s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [16m30s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [16m40s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [16m50s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [17m0s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [17m10s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [17m20s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [17m30s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [17m40s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [17m50s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [18m0s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [18m10s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [18m20s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [18m30s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [18m40s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [18m50s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [19m0s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [19m10s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [19m20s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [19m30s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [19m40s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [19m50s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [20m0s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [20m10s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [20m20s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [20m30s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [20m40s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [20m50s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [21m0s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [21m10s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [21m20s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [21m30s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [21m40s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [21m50s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [22m0s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [22m10s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [22m20s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [22m30s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [22m40s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [22m50s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [23m0s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [23m10s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [23m20s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [23m30s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [23m40s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [23m50s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [24m0s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [24m10s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [24m20s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [24m30s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [24m40s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [24m50s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [25m0s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [25m10s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [25m20s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [25m30s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [25m40s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [25m50s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [26m0s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [26m10s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [26m20s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [26m30s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [26m40s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [26m50s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [27m0s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [27m10s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [27m20s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [27m30s elapsed]
aws_eks_identity_provider_config.named_test_resource: Still creating... [27m40s elapsed]
aws_eks_identity_provider_config.named_test_resource: Creation complete after 27m48s [id=turbottest32066:turbottest32066]

Warning: Deprecated Resource

The null_data_source was historically used to construct intermediate values to
re-use elsewhere in configuration, the same can now be achieved using locals


Apply complete! Resources: 8 added, 0 changed, 0 destroyed.

Outputs:

account_id = "013122550996"
cluster_name = "turbottest32066"
identity = tolist([
  {
    "oidc" = tolist([
      {
        "issuer" = "https://oidc.eks.us-east-1.amazonaws.com/id/D7BC8F3C8E5982FC1FC112CC8C5E24E7"
      },
    ])
  },
])
identity_provider_config_type = "oidc"
resource_aka = "arn:aws:eks:us-east-1:013122550996:identityproviderconfig/turbottest32066/oidc/turbottest32066/20bd765b-a9f4-a0e1-2eb1-5fe499d3cdd4"
resource_id = "turbottest32066:turbottest32066"
resource_name = "turbottest32066"
role_arn = "arn:aws:iam::013122550996:role/turbottest32066"
status = "ACTIVE"

Running SQL query: test-get-query.sql
[
  {
    "arn": "arn:aws:eks:us-east-1:013122550996:identityproviderconfig/turbottest32066/oidc/turbottest32066/20bd765b-a9f4-a0e1-2eb1-5fe499d3cdd4",
    "cluster_name": "turbottest32066",
    "name": "turbottest32066",
    "type": "oidc"
  }
]
✔ PASSED

Running SQL query: test-list-query.sql
[
  {
    "arn": "arn:aws:eks:us-east-1:013122550996:identityproviderconfig/turbottest32066/oidc/turbottest32066/20bd765b-a9f4-a0e1-2eb1-5fe499d3cdd4",
    "name": "turbottest32066"
  }
]
✔ PASSED

Running SQL query: test-not-found-query.sql
null
✔ PASSED

Running SQL query: test-turbot-query.sql
[
  {
    "account_id": "013122550996",
    "akas": [
      "arn:aws:eks:us-east-1:013122550996:identityproviderconfig/turbottest32066/oidc/turbottest32066/20bd765b-a9f4-a0e1-2eb1-5fe499d3cdd4"
    ],
    "name": "turbottest32066",
    "tags": {
      "Name": "turbottest32066"
    },
    "title": "turbottest32066"
  }
]
✔ PASSED

POSTTEST: tests/aws_eks_identity_provider_config

TEARDOWN: tests/aws_eks_identity_provider_config

SUMMARY:

1/1 passed.
```
</details>

# Example query results
<details>
  <summary>Results</summary>
  
### Basic Info

```sql
select
  name,
  client_id,
  arn,
  cluster_name,
  tags,
  status
from
  aws_eks_identity_provider_config;
```

```
+---------------+-----------+--------------------------------------------------------------------------------------------------------------------------------+--------------+------+----------+
| name          | client_id | arn                                                                                                                            | cluster_name | tags | status   |
+---------------+-----------+--------------------------------------------------------------------------------------------------------------------------------+--------------+------+----------+
| test-identity | test      | arn:aws:eks:us-east-2:986325076436:identityproviderconfig/test_cluster/oidc/test-identity/f8bd8388-6683-996d-add2-217c81de0739 | test_cluster | {}   | CREATING |
+---------------+-----------+--------------------------------------------------------------------------------------------------------------------------------+--------------+------+----------+
```

### List OIDC type Identity provider config

```sql
select
  name,
  arn,
  cluster_name,
  type
from
  aws_eks_identity_provider_config
where 
  type = 'oidc';
```

```
+---------------+--------------------------------------------------------------------------------------------------------------------------------+--------------+------+
| name          | arn                                                                                                                            | cluster_name | type |
+---------------+--------------------------------------------------------------------------------------------------------------------------------+--------------+------+
| test-identity | arn:aws:eks:us-east-2:986325076436:identityproviderconfig/test_cluster/oidc/test-identity/f8bd8388-6683-996d-add2-217c81de0739 | test_cluster | oidc |
+---------------+--------------------------------------------------------------------------------------------------------------------------------+--------------+------+
```

</details>

### CODE REVIEW

File: aws-test/tests/aws_eks_identity_provider_config/variables.tf [hunk 1]
```diff
@@ -0,0 +1,171 @@
+
```
Comment (khushboo9024) [inline aws-test/tests/aws_eks_identity_provider_config/variables.tf:1]: Remove this extra line.

File: aws/table_aws_eks_identity_provider_config.go [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+	"github.com/turbot/steampipe-plugin-sdk/plugin"
+	"github.com/turbot/steampipe-plugin-sdk/plugin/transform"
+
+	"github.com/aws/aws-sdk-go/service/eks"
+)
+
+// type ClusterNameWithConfig struct {
+// 	ClusterName *string
+// 	*eks.IdentityProviderConfig
+// }
```
Comment (khushboo9024) [inline aws/table_aws_eks_identity_provider_config.go:16]: Remove this commented lines.

File: aws/table_aws_eks_identity_provider_config.go [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+	return &plugin.Table{
+		Name:        "aws_eks_identity_provider_config",
+		Description: "AWS EKS Identity Provider Config",
+		Get: &plugin.GetConfig{
+			KeyColumns: plugin.AllColumns([]string{"name", "type", "cluster_name"}),
+			Hydrate:    getEksIdentityProviderConfig,
+		},
+		List: &plugin.ListConfig{
+			ParentHydrate: listEksClusters,
+			Hydrate:       listEksIdentityProviderConfig,
```
Comment (khushboo9024) [inline aws/table_aws_eks_identity_provider_config.go:36]:
```suggestion
			Hydrate:       listEksIdentityProviderConfigs,
```

File: aws/table_aws_eks_identity_provider_config.go [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
+		},
+		GetMatrixItem: BuildRegionList,
+		Columns: awsRegionalColumns([]*plugin.Column{
+			{
+				Name:        "name",
+				Description: "The name of the identity provider configuration.",
+				Type:        proto.ColumnType_STRING,
+				// Transform:   transform.FromP(getAwsEksProviderConfigNameAndType, "Name"),
```
Comment (khushboo9024) [inline aws/table_aws_eks_identity_provider_config.go:44]: Remmove commented lines.

File: aws/table_aws_eks_identity_provider_config.go [hunk 5]
```diff
@@ -0,0 +1,228 @@
...
+			},
+			{
+				Name:        "username_claim",
+				Description: "The JSON Web token (JWT) claim that is used as the username.",
+				Type:        proto.ColumnType_STRING,
+				Hydrate:     getEksIdentityProviderConfig,
+			},
+			{
+				Name:        "username_prefix",
+				Description: "The JSON Web token (JWT) claim that is used as the username.",
```
Comment (khushboo9024) [inline aws/table_aws_eks_identity_provider_config.go:109]: Ordering of columns is not correct, JSON columns should come after the string columns.

File: aws/table_aws_eks_identity_provider_config.go [hunk 6]
```diff
@@ -0,0 +1,228 @@
...
+	var clusterName, providerConfigName, providerConfigType string
+	if h.Item != nil {
+		clusterName = *h.Item.(*IdentityProviderConfig).ClusterName
+		providerConfigName = *h.Item.(*IdentityProviderConfig).Name
+		providerConfigType = *h.Item.(*IdentityProviderConfig).Type
+	} else {
+		clusterName = d.KeyColumnQuals["cluster_name"].GetStringValue()
+		providerConfigName = d.KeyColumnQuals["name"].GetStringValue()
+		providerConfigType = d.KeyColumnQuals["type"].GetStringValue()
+
```
Comment (khushboo9024) [inline aws/table_aws_eks_identity_provider_config.go:183]:
```suggestion
```

File: aws/table_aws_eks_identity_provider_config.go [hunk 7]
```diff
@@ -0,0 +1,228 @@
...
+// 		return nil, nil
+// 	}
+// 	if d.Param.(string) == "Name" {
+// 		return configProvider.Name, nil
+// 	} else if d.Param.(string) == "Type" {
+// 		return configProvider.Type, nil
+// 	} else {
+// 		return nil, nil
+// 	}
+// }
```
Comment (khushboo9024) [inline aws/table_aws_eks_identity_provider_config.go:228]: Do we need this?

File: aws/table_aws_eks_identity_provider_config.go [hunk 8]
```diff
@@ -0,0 +1,228 @@
...
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     getEksIdentityProviderConfig,
+			},
+			{
+				Name:        "status",
+				Description: "The status of the OIDC identity provider.",
+				Type:        proto.ColumnType_STRING,
+				Hydrate:     getEksIdentityProviderConfig,
+			},
+			{
```
Comment (khushboo9024) [inline aws/table_aws_eks_identity_provider_config.go:94]: `ClientId` cloumn is missing.

**Review by khushboo9024 — CHANGES_REQUESTED** [review]
Add sample queries result in the logs.

File: aws-test/tests/aws_eks_identity_provider_config/variables.tf (hunk 1 above)
Comment (ParthaI) [reply aws-test/tests/aws_eks_identity_provider_config/variables.tf:1]: Removed extra line

File: aws/table_aws_eks_identity_provider_config.go (hunk 2 above)
Comment (ParthaI) [reply aws/table_aws_eks_identity_provider_config.go:16]: Removed the commented lines

File: aws/table_aws_eks_identity_provider_config.go (hunk 3 above)
Comment (ParthaI) [reply aws/table_aws_eks_identity_provider_config.go:36]: Renamed the function name

File: aws/table_aws_eks_identity_provider_config.go (hunk 4 above)
Comment (ParthaI) [reply aws/table_aws_eks_identity_provider_config.go:44]: Removed

File: aws/table_aws_eks_identity_provider_config.go (hunk 7 above)
Comment (ParthaI) [reply aws/table_aws_eks_identity_provider_config.go:228]: removed the unnecessary commented codes

File: aws/table_aws_eks_identity_provider_config.go (hunk 6 above)
Comment (ParthaI) [reply aws/table_aws_eks_identity_provider_config.go:183]: Removed the extra line

File: aws/table_aws_eks_identity_provider_config.go (hunk 5 above)
Comment (ParthaI) [reply aws/table_aws_eks_identity_provider_config.go:109]: Managed the column ordering

File: aws/table_aws_eks_identity_provider_config.go (hunk 8 above)
Comment (ParthaI) [reply aws/table_aws_eks_identity_provider_config.go:94]: Added the `ClientId` Column

**Review by khushboo9024 — APPROVED** [review]

**Review by rajlearner17 — APPROVED** [review]
LGTM

**Pull request merged**